### ISSUE TITLE
UnusedImports does not report unused static imports of enums when they are used as switch targets

### ISSUE THREAD

**Issue by ksobolew** [checkstyle/checkstyle#13361]
I have read check documentation: https://checkstyle.org/checks/imports/unusedimports.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
kudivuhadi@MBP-Krzysztof scratches %  javac scratch.java                                           
kudivuhadi@MBP-Krzysztof scratches %  cat config.xml 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="UnusedImports" />
    </module>
</module>

kudivuhadi@MBP-Krzysztof scratches %  cat scratch.java 
import java.lang.annotation.ElementType;

import static java.lang.annotation.ElementType.CONSTRUCTOR;
import static java.lang.annotation.ElementType.FIELD;
import static java.lang.annotation.ElementType.METHOD;

class Dummy
{
    public static void main(String[] args)
    {
        ElementType test = METHOD;
        String result = switch (test)
        {
            case METHOD -> "m";
            case FIELD -> "f";
            default -> "o";
        };
    }
}

kudivuhadi@MBP-Krzysztof scratches %  RUN_LOCALE="-Duser.language=en -Duser.country=US"             
kudivuhadi@MBP-Krzysztof scratches %  java $RUN_LOCALE -jar ~/Downloads/checkstyle-10.12.1-all.jar \
   -c config.xml scratch.java

Starting audit...
[ERROR] /Users/kudivuhadi/Library/Application Support/JetBrains/IntelliJIdea2023.2/scratches/scratch.java:3:15: Unused import - java.lang.annotation.ElementType.CONSTRUCTOR. [UnusedImports]
Audit done.
Checkstyle ends with 1 errors.
```

---

Expected:

```
Starting audit...
[ERROR] /Users/kudivuhadi/Library/Application Support/JetBrains/IntelliJIdea2023.2/scratches/scratch.java:3:15: Unused import - java.lang.annotation.ElementType.CONSTRUCTOR. [UnusedImports]
[ERROR] /Users/kudivuhadi/Library/Application Support/JetBrains/IntelliJIdea2023.2/scratches/scratch.java:4:15: Unused import - java.lang.annotation.ElementType.FIELD. [UnusedImports]
Audit done.
Checkstyle ends with 2 errors.
```
The imports for `CONSTRUCTOR` and `FIELD` are unsed -- IntelliJ will remove them also. 

This is in fact very similar to #12923, but for enum constants and `switch`.

The Java file contains three static imports of an enum (`java.lang.annotation.ElementType`) constants: `CONSTRUCTOR`, `FIELD`, and `METHOD`. There is also a `switch` which has branches for `METHOD` and `FIELD`. The thing with switches is that the branch targets are "obvious" from the type of the value switched on, so they are not typical constant references and do not need to be neither fully qualified nor static-imported. So the static imports of `CONSTRUCTOR` and `FIELD` are not needed (`METHOD` is referenced directly in an assignment, so it's not unused). But Checkstyle reports only `CONSTRUCTOR`, which is neither referenced directly nor used as a branch target in the switch, and `FIELD` is a false negative.

I'll try to be a little more clear...

When you have a `switch` on enum, the switch branch targets are constants of that enum. They can't be anything else, because it's obvious from context what they have to be - in a sense they are in a separate namespace from statically-imported constant references. For that reason, they do not need to be imported. @nrmancuso demonstrates that the code compiles just fine without the two imports, and @electrum points out that IntelliJ IDEA correctly marks them as unused, too.

**Comment by romani** [comment]
@ksobolew , please fix link `I have read check documentation:` to make it clear you did read doc.
It also helps any contributor in the world to understand and start working on issue easier.

**Comment by romani** [comment]
I do not very clearly understand explanation.
 This is need to wait for laptop time to experience this new syntax.

**Comment by electrum** [comment]
The imports for `CONSTRUCTOR` and `FIELD` are unsed -- IntelliJ will remove them. This import behavior for enums is the same with the old style switch:
```java
import java.lang.annotation.ElementType;

import static java.lang.annotation.ElementType.CONSTRUCTOR; // unused
import static java.lang.annotation.ElementType.FIELD; // unused
import static java.lang.annotation.ElementType.METHOD;

class Dummy
{
    public static void main(String[] args)
    {
        ElementType test = METHOD;
        String result;
        switch (test) {
            case METHOD:
                result = "m";
                break;
            case FIELD:
                result = "f";
                break;
            default:
                result = "o";
                break;
        }
    }
}

**Comment by romani** [comment]
We to prove that code that initially reported can compile without `java.lang.annotation.ElementType.CONSTRUCTOR`

**Comment by romani** [comment]
I can not find java compiler online for most recent version

**Comment by nrmancuso** [comment]
```
➜  src cat Test.java
import java.lang.annotation.ElementType;

import static java.lang.annotation.ElementType.CONSTRUCTOR; // unused
import static java.lang.annotation.ElementType.FIELD; // unused
import static java.lang.annotation.ElementType.METHOD;

class Test
{
    public static void main(String[] args)
    {
        ElementType test = METHOD;
        String result = switch (test)
        {
            case METHOD -> "m";
            case FIELD -> "f";
            default -> "o";
        };
    }
}
➜  src javac --version
javac 19.0.1
➜  src javac Test.java
➜  src echo $?
0

```

**Comment by romani** [comment]
@ksobolew,  please explain why you think this is issue?

**Comment by ksobolew** [comment]
I'll try to be a little more clear...

When you have a `switch` on enum, the switch branch targets are constants of that enum. They can't be anything else, because it's obvious from context what they have to be - in a sense they are in a separate namespace from statically-imported constant references. For that reason, they do not need to be imported. @nrmancuso demonstrates that the code compiles just fine without the two imports, and @electrum points out that IntelliJ IDEA correctly marks them as unused, too.

**Comment by romani** [comment]
@ksobolew,

>For that reason, they do not need to be importe

Please update issue description and clearly define what is expected out of CLI ( amount of violations and exact line numbers).
Example: https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F

Thanks in advance.

**Comment by romani** [comment]
I agree to approve this issue.
@rnveach , please review and mark it by approved label if you agree.

**Comment by romani** [comment]
Yes, you are right.

From issue description:
>The thing with switches is that the branch targets are "obvious" from the type of the value switched on, so they are not typical constant references and do not need to be neither fully qualified nor static-imported.

More specifically:
>branch targets are "obvious" from the type

Unfortunately, we are not type-aware tool. Link to limits is above.

We can not do violation von this reliably, as type name can be hidden by `var` or any other obscure logic when type is not visible in code, but yes compiler knows all .
